controlplane/kubeadm/controllers/scale.go

```diff
@@ -135,5 +135,15 @@ func selectMachineForScaleDown(controlPlane *internal.ControlPlane, outdatedMach
 	if outdatedMachines.Len() > 0 {
 		machines = outdatedMachines
 	}
+
+	withDeleteAnnotation, err := controlPlane.MachineWithDeleteAnnotation(machines)
+	if err != nil {
```

Comment:
the func MachineWithDeleteAnnotation should not block the selection process if it returns a `nil` machine. I believe if it cannot find a machine to delete, we should move to the func `MachineInFailureDomainWithMostMachines(machines)`